File: internal/store/outreach_template.go
package store

import "database/sql"

// outreachTemplateKey is the fixed primary key of the singleton email-template
// row; followupTemplateKey is the parallel singleton for the follow-up template
// (M53) — both live in outreach_template, keyed apart.
const (
	outreachTemplateKey = "default"
	followupTemplateKey = "followup"
)

// GetOutreachTemplate returns the saved email template, or "" when none has been
// saved (the caller falls back to the compiled-in default).
func (db *DB) GetOutreachTemplate() (string, error) {
	var c string
	err := db.QueryRow("SELECT content FROM outreach_template WHERE key = ?", outreachTemplateKey).Scan(&c)
	if err == sql.ErrNoRows {
		return "", nil
	}
	return c, err
}

// PutOutreachTemplate upserts the singleton template row.
func (db *DB) PutOutreachTemplate(content string) error {
	return db.putTemplate(outreachTemplateKey, content)
}

// GetFollowupTemplate returns the saved follow-up template, or "" when none has
// been saved (the caller falls back to the compiled-in default).
func (db *DB) GetFollowupTemplate() (string, error) {
	var c string
	err := db.QueryRow("SELECT content FROM outreach_template WHERE key = ?", followupTemplateKey).Scan(&c)
	if err == sql.ErrNoRows {
		return "", nil
	}
	return c, err
}

// PutFollowupTemplate upserts the singleton follow-up template row.
func (db *DB) PutFollowupTemplate(content string) error {
	return db.putTemplate(followupTemplateKey, content)
}

func (db *DB) putTemplate(key, content string) error {
	const q = `
INSERT INTO outreach_template (key, content, updated_at)
VALUES (?, ?, CURRENT_TIMESTAMP)
ON CONFLICT(key) DO UPDATE SET content = excluded.content, updated_at = CURRENT_TIMESTAMP`
	_, err := db.Exec(q, key, content)
	return err
}

File: scout/store/outreach_template.py
"""Singleton email + follow-up template rows."""

from __future__ import annotations

import sqlite3

# Both live in outreach_template, keyed apart.
_OUTREACH_TEMPLATE_KEY = "default"
_FOLLOWUP_TEMPLATE_KEY = "followup"


def get_outreach_template(con: sqlite3.Connection) -> str:
    """The saved email template, or "" when none has been saved."""
    row = con.execute(
        "SELECT content FROM outreach_template WHERE key = ?", (_OUTREACH_TEMPLATE_KEY,)
    ).fetchone()
    return row[0] if row is not None else ""


def put_outreach_template(con: sqlite3.Connection, content: str) -> None:
    """Upsert the singleton email-template row."""
    _put_template(con, _OUTREACH_TEMPLATE_KEY, content)


def get_followup_template(con: sqlite3.Connection) -> str:
    """The saved follow-up template, or "" when none has been saved."""
    row = con.execute(
        "SELECT content FROM outreach_template WHERE key = ?", (_FOLLOWUP_TEMPLATE_KEY,)
    ).fetchone()
    return row[0] if row is not None else ""


def put_followup_template(con: sqlite3.Connection, content: str) -> None:
    """Upsert the singleton follow-up template row."""
    _put_template(con, _FOLLOWUP_TEMPLATE_KEY, content)


def _put_template(con: sqlite3.Connection, key: str, content: str) -> None:
    con.execute(
        """
INSERT INTO outreach_template (key, content, updated_at)
VALUES (?, ?, CURRENT_TIMESTAMP)
ON CONFLICT(key) DO UPDATE SET content = excluded.content, updated_at = CURRENT_TIMESTAMP""",
        (key, content),
    )
